### PR TITLE
Implement proper waiting for resources in fast-integration

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- migrate from `await sleep(X)` to promise retries

**Related issue(s)**

### CODE REVIEW

File: tests/fast-integration/test/k8s-test.js [hunk 1]
```diff
@@ -55,71 +76,91 @@ describe("Commerce Mock tests", function () {
...
+      response = await retryPromise(
+        async () => {
+          console.log(`Calling https://${mockHost}/local/apis`);
+          return axios.get(`https://${mockHost}/local/apis`).then((res) => {
+            expect(res.data).to.have.lengthOf(2);
+            expect(res.data[0].provider).not.to.be.empty;
+            return res;
+          });
+        },
+        20,
```
Comment (pbochynski) [inline tests/fast-integration/test/k8s-test.js:136]: This timeout is too short. Sometimes it is more than 1 minute (Gardener Azure). 2 minutes should be enough.

**Review by pbochynski — APPROVED** [review]
Works on my cluster 👍

File: tests/fast-integration/test/k8s-test.js [hunk 2]
```diff
@@ -55,71 +76,91 @@ describe("Commerce Mock tests", function () {
...
-        "networking.istio.io",
-        "v1beta1",
-        "mocks",
-        "virtualservices",
-        "true",
-        undefined,
-        undefined,
-        "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
+      console.log("Waiting for virtual service to be ready...");
+      virtualservice = await retryPromise(
```
Comment (pbochynski) [inline tests/fast-integration/test/k8s-test.js:92]: I would refactor the test to wrap long API invocations in some functions, example:
```
function getMockVirtualHost() {
  return k8sCRDApi
            .listNamespacedCustomObject(
              "networking.istio.io",
              "v1beta1",
              "mocks",
              "virtualservices",
              "true",
              undefined,
              undefined,
              "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
            )
            .then((res) => {
              expect(res.body.items).to.have.lengthOf(1);
              expect(res.body.items[0]).to.have.nested.property("spec.hosts");
              expect(res.body.items[0].spec.hosts).to.have.lengthOf(1);
              expect(res.body.items[0].spec.hosts[0]).not.to.be.empty;
              return res;
            });
}

...


    let virtualservice;
    try {
      console.log("Waiting for virtual service to be ready...");
      virtualservice = await retryPromise(getMockVirtualHost, 10, 1000 );
    } catch (err) {
      expect(err.body.message).to.be.empty;
    }
```

**Comment by aerfio** [comment]
I'll answer your PR comments in my next PR https://github.com/kyma-project/kyma/pull/10214

**Pull request merged**